src/dns/local.rs

```diff
@@ -0,0 +1,134 @@
+use std::collections::HashMap;
+use std::net::IpAddr;
+use wildmatch::WildMatch;
+use crate::dns::DNSEntry;
+
+/// Local override table that can be used instead of using /etc/hosts or similar 3rd party dns system.
+pub struct LocalOverrideTable {
+    /// Entries in the local override table. First iteration of the resolver: a simple list that
+    /// will be queried O(n). Later iterations will use a more efficient data structure like a tree
+    /// per domain part (e.g. com, org, net, etc.)
+    entries: Vec<DNSEntry>,
+    round_robin_ptrs: HashMap<String, usize>,
+}
+
+impl Default for LocalOverrideTable {
+    fn default() -> LocalOverrideTable {
+        LocalOverrideTable {
+            entries: Vec::new(),
+            round_robin_ptrs: HashMap::new(),
```

Comment:
if a domain has multiple ips, we store a round robin pointer so the next call to resolve will return the next ip in the list.